sweep/sweeper.go

```diff
@@ -464,27 +521,25 @@ func (s *UtxoSweeper) scheduleSweep(currentHeight int32) error {
 		return nil
 	}
 
-	// Retrieve fee estimate for input filtering and final tx fee
-	// calculation.
-	satPerKW, err := s.cfg.FeeEstimator.EstimateFeePerKW(
-		s.cfg.SweepTxConfTarget,
-	)
-	if err != nil {
-		return fmt.Errorf("estimate fee: %v", err)
-	}
+	// We'll only start our timer once we have inputs we're able to sweep.
+	startTimer := false
+	for feePref := range s.inputsByFeePref {
+		// Examine pending inputs and try to construct lists of inputs.
+		inputLists, err := s.getInputLists(currentHeight, feePref)
+		if err != nil {
+			return fmt.Errorf("get input lists: %v", err)
+		}
 
-	// Examine pending inputs and try to construct lists of inputs.
-	inputLists, err := s.getInputLists(currentHeight, satPerKW)
-	if err != nil {
-		return fmt.Errorf("get input lists: %v", err)
-	}
+		log.Infof("Sweep candidates at height=%v, fee_preference=%v, "+
+			"yield %v distinct txns", currentHeight, feePref,
+			len(inputLists))
 
-	log.Infof("Sweep candidates at height=%v, yield %v distinct txns",
-		currentHeight, len(inputLists))
+		if len(inputLists) != 0 {
+			startTimer = true
```

Comment:
I think timers should be separate per fee pref. The idea is to allow time for more inputs to come in, but if they are in a different bucket, that won't work as expected.

Reply:
Isn't that the case with this change already? If any of the buckets has inputs, then we'll start the timer to collect more.

Reply:
There is still a single timer. The idea of the timer is that once the first input for a tx comes in, we start a timer to accumulate more inputs for that tx. Because of the buckets, we are now accumulating inputs for multiple txes in parallel. So suppose input A with fee rate 10 comes in. This will start the timer. Then after 29 seconds, input B with fee rate 15 comes in. One second later, two txes are published, but the fee rate 15 tx didn't get its 30 seconds to accumulate more inputs.

Reply:
But it could also be argued that all inputs typically come in in a 30 second window after a block is mined anyway, regardless of the fee rate. In that case it wouldn't matter whether you have one or multiple timers, as they all expire at the same moment.

Reply:
> But it could also be argued that all inputs typically come in in a 30 second window after a block is mined anyway, regardless of the fee rate. In that case it wouldn't matter whether you have one or multiple timers, as they all expire at the same moment.

Yeah this is how I was thinking of the timer. Seems fine to leave as is then?

Reply:
Yes I think it is fine